elks/arch/i86/drivers/char/crtc-6845.c

```diff
@@ -0,0 +1,69 @@
+#include <arch/io.h>
+#include <linuxmt/config.h>
+#include <linuxmt/init.h>
+#include <linuxmt/memory.h>
+#include "crtc-6845.h"
+
+/*
+ * https://stanislavs.org/helppc/6845.html
+ * https://www.seasip.info/VintagePC/index.html
+ *
+ *     I/O          VidBase  VidLen       Notes
+ * MDA: 0x3B0-0x3BB 0xB0000  4k  (0x1000)
+ * CGA: 0x3D0-0x3DC 0xB8000  16k (0x4000) note: Address not fully decoded, fb is also at 0xBC000
+ * EGA: 0x3C0-0x3CF ???????  ?????
+ * VGA: ?????
+ */
+
+#define CRTC_INDX 0x0
+#define CRTC_DATA 0x1
+#define CRTC_MODE 0x4
+#define CRTC_CSEL 0x5
+#define CRTC_STAT 0x6
+
+/* Check to see if this CRTC is present */
+char INITPROC crtc_probe(unsigned short crtc_base)
```

Comment:
I can't recommend using `char` either as a return value or an argument: there's too many problems with it. For instance, the compiler will generate `CBW` (convert byte to word) instead of just using AX when using the return value of this function (needlessly).

Reply:
Good to know! I need to get into the habit of checking out the generated assembly more often.